LocalKeyClientEncryption/LocalKeyClientEncryption.cs

```diff
@@ -0,0 +1,309 @@
+        * Microsoft Managed Keys, Customer Managed Keys, and Customer Provided Keys
+        *
+        *
+        * NOTE: This program requires the following to be stored in the App.Config file:
+        * TENANT_ID
+        * CLIENT_ID
+        * CLIENTSECRET
+        * Azure Subscription ID
+        * Resource Group Name
+        * Storage Account Name
+        * Storage Account Connection String
+        * Key Vault Name
+        * Customer Provided Key Used for Client Side Encryption 
+        * Key Wrap Algorithm Used for Client Side Encryption ("ExampleAlgorithm" in SampleKeyEncryptionKey)
+        *  
+        * MUST MATCH PARAMETERS USED TO ENCRYPT CLIENT SIDE ENCRYPTED DATA
+        */
+        static void Main()
+        {
+            var config = ConfigurationManager.AppSettings;
+
+            //Credentials of Service Principal
+            TokenCredential credential =
+                new ClientSecretCredential(
+                    config["tenantId"],
+                    config["clientId"],
+                    config["clientSecret"]
+                    );
+
+            string keyVaultName = config["keyVaultName"];
+            string connectionString = config["connectionString"];
+
+
+            //Edit the Following as Needed
+            //Program Creates the Following using the Provided Names
+            string containerName = "example" + Guid.NewGuid().ToString();
+            string fileName = "example.txt";
+            string keyVaultKeyName = "testKey";
+            string encryptionScopeName = "myencryption";
+            string customerProvidedKey = "dfD3Jb#6htqfpoj@gGpomDAv21035%21";     //Key used for Customer Provided Key Server Side Encryption
+
+            byte[] localKeyBytes = ASCIIEncoding.UTF8.GetBytes(customerProvidedKey);
+
+
+
+
+            //File Path for local file used to upload and reupload
+            string localPath = "./data/";
+            Directory.CreateDirectory(localPath);
+            string localFilePath = Path.Combine(localPath, fileName);
+
+            //Creating Key Encryption Key object for Client Side Encryption
+            SampleKeyEncryptionKey keyEncryption = new SampleKeyEncryptionKey(config["clientSideCustomerProvidedKey"]);
+
+            //Set up Client Side Encryption Options used for Client Side Encryption
+            ClientSideEncryptionOptions clientSideOptions = new ClientSideEncryptionOptions(ClientSideEncryptionVersion.V1_0)
+            {
+                KeyEncryptionKey = keyEncryption,
+                KeyWrapAlgorithm = config["keyWrapAlgorithm"]
+            };
+
+            //Create Blob Service Client
+            BlobServiceClient blobServiceClient = new BlobServiceClient(connectionString);
+
+            //Run Setup Function that creates and example container and blob
+            SetupForExample(blobServiceClient, containerName, fileName, localFilePath, clientSideOptions);
+
+            Console.Write("Press Enter to change to SSE");

```

Comment:
Could we just power through this and allow them to go from CSE to SSE, also maybe we should change this so they can enter what kind of encryption they want to do